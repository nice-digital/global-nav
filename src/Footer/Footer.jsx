import React, { Component } from "react";
import PropTypes from "prop-types";

import NiceLogo from "@nice-digital/icons/lib/Logo";

import Social from "./Social";
import Legal from "./Legal";
import Services from "./Services";
import Pages from "./Pages";
import styles from "./Footer.module.scss";
import TrackedLink from "../TrackedLink";
import { footerClickEventAction } from "../tracker";
import { BackToTop } from "./BackToTop/BackToTop";

export class Footer extends Component {
	render() {
		return (
			<>
				<BackToTop />
				<footer className={styles.footer} data-tracking="Global footer">
					<div className={styles.container}>
						<TrackedLink
							eventAction={footerClickEventAction}
							eventLabel="Logo"
							href="https://www.nice.org.uk/"
							className={styles.logo}
							aria-label="Go to NICE home page"
						>
							<NiceLogo />
						</TrackedLink>
						<Services service={this.props.service} />
						<Pages service={this.props.service} />
						<Social />
					</div>
					<Legal />
				</footer>
			</>
		);
	}
}

Footer.propTypes = {
	service: PropTypes.string,
};

export default Footer;
