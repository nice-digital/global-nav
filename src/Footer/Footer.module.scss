@use 'sass:math';
@import '~@nice-digital/nds-core/scss/core';
@import 'footer-settings';
@import '../Common.module.scss';

.footer {
  @include nds-links-inverse;
  background: $colour-footer-background;
  box-sizing: border-box;
  color: $colour-footer-text;
  contain-intrinsic-size: 100vw rem(1090); /* stylelint-disable-line */
  content-visibility: auto; /* stylelint-disable-line */
  font-family: unquote($nds-font-family-sans);
  margin: rem($nds-spacing-large 0 0 0);
  position: relative;
  z-index: 0;

  *,
  *:before,
  *:after {
    box-sizing: inherit;
  }

  a {
    font-weight: normal; // To override Pathways global anchor style
  }

  // Estimates of intrinsic height across breakpoints
  @include mq($from: xs) {
    contain-intrinsic-height: rem(930);
  }
  @include mq($from: sm) {
    contain-intrinsic-height: rem(540);
  }
  @include mq($from: md) {
    contain-intrinsic-height: rem(450);
  }
  @include mq($from: lg) {
    contain-intrinsic-height: rem(495);
  }
}

.container {
  @include nds-container;
  @include nds-clearfix;
  @include nds-hide-print;
  padding: rem($nds-spacing-medium 0);
}

.logo {
  $logo-width: 80px;
  $logo-aspect-ratio: math.div(512, 1410);
  display: block;
  height: 0;
  margin: rem(0 0 $nds-spacing-large 0);
  padding: 0;
  padding-bottom: $logo-width * $logo-aspect-ratio; // Aspect ratio of NICE logo SVG
  position: relative;
  width: $logo-width;

  svg {
    display: block;
    height: 100% !important;
    left: 0;
    position: absolute;
    top: 0;
    width: 100% !important;
  }
}

@include mq($from: md) {
  .logo {
    margin-bottom: rem($nds-spacing-medium);
  }
}
