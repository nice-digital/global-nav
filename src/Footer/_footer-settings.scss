@use '@nice-digital/nds-core/scss/colours';

$colour-footer-background: colours.$nice-black;
$colour-footer-text: colours.$nice-white;
