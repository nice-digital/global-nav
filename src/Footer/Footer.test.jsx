import React from "react";

import Footer from "./Footer";
import { render } from "@testing-library/react";

describe("Footer", () => {
	it("Matches snapshot", () => {
		const { container } = render(<Footer service="cks" />);
		expect(container).toMatchSnapshot();
	});
});
