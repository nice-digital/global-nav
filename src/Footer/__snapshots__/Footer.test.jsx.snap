// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Matches snapshot 1`] = `
<div>
  <footer
    class="footer"
    data-tracking="Global footer"
  >
    <div
      class="container"
    >
      <a
        aria-label="Go to NICE home page"
        class="logo"
        href="https://www.nice.org.uk/"
      >
        <svg
          aria-hidden="true"
          class="icon"
          fill="none"
          focusable="false"
          height="1em"
          viewBox="0 0 1381 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M396.692 37.908v436.036h-79.615L127.42 199.501h-3.25v274.443H32V37.908h80.944l188.181 274.295h3.84V37.908h91.727ZM552.082 37.908h-92.17v436.036h92.17V37.908ZM1000.08 190.639h-93.202c-1.773-12.112-5.17-22.747-10.488-32.2-5.317-9.454-11.964-17.43-20.236-24.225-8.271-6.647-17.725-11.816-28.36-15.361-10.782-3.545-22.304-5.318-34.859-5.318-22.599 0-42.245 5.613-58.936 16.691-16.691 11.078-29.689 27.326-38.995 48.596-9.158 21.27-13.884 46.972-13.884 77.252 0 30.28 4.579 57.163 13.884 78.285 9.306 21.123 22.304 37.075 38.995 47.858 16.691 10.783 36.189 16.1 58.198 16.1 12.407 0 23.781-1.625 34.416-4.874 10.635-3.25 19.94-7.976 28.212-14.328 8.272-6.351 15.066-14.032 20.531-23.042 5.466-9.011 9.306-19.498 11.374-31.019l93.204.443c-2.363 19.941-8.419 39.29-18.02 57.754-9.601 18.463-22.452 35.007-38.552 49.63-16.101 14.475-35.303 25.997-57.459 34.416C863.747 475.716 838.636 480 810.572 480c-38.848 0-73.559-8.863-104.135-26.44-30.575-17.577-54.652-43.131-72.377-76.365-17.725-33.382-26.588-73.707-26.588-121.121 0-47.415 9.011-88.034 26.883-121.417 17.873-33.382 42.097-58.788 72.821-76.365C737.899 40.715 772.315 32 810.572 32c25.258 0 48.744 3.545 70.309 10.635 21.713 7.09 40.915 17.43 57.606 31.019 16.691 13.589 30.428 30.132 40.916 49.778 10.635 19.645 17.281 41.949 20.383 67.207M1053.55 473.944V37.908h293.79v76.07h-201.62v103.839h186.56v76.07h-186.56v104.135h202.51v75.922h-294.68Z"
            fill="currentColor"
          />
        </svg>
      </a>
      <nav
        aria-label="Our services"
        class="wrapper"
      >
        <ul
          class="list"
        >
          <li>
            <a
              class="link"
              href="https://www.nice.org.uk/guidance"
            >
              Guidance
            </a>
          </li>
          <li>
            <a
              class="link"
              href="https://www.nice.org.uk/standards-and-indicators"
            >
              Standards and indicators
            </a>
          </li>
          <li>
            <a
              class="link"
              href="https://cks.nice.org.uk/"
            >
              <abbr
                title="Clinical Knowledge Summaries"
              >
                Clinical Knowledge Summaries (CKS)
                 
                <span
                  class="visuallyHidden"
                >
                  Clinical Knowledge Summaries
                </span>
              </abbr>
              <span
                aria-hidden="true"
                class="tooltip"
              >
                Clinical Knowledge Summaries
              </span>
            </a>
          </li>
          <li>
            <a
              class="link"
              href="https://bnf.nice.org.uk/"
            >
              <abbr
                title="British National Formulary"
              >
                British National Formulary (BNF)
                 
                <span
                  class="visuallyHidden"
                >
                  British National Formulary
                </span>
              </abbr>
              <span
                aria-hidden="true"
                class="tooltip"
              >
                British National Formulary
              </span>
            </a>
          </li>
          <li>
            <a
              class="link"
              href="https://bnfc.nice.org.uk/"
            >
              <abbr
                title="British National Formulary for Children"
              >
                British National Formulary for Children (BNFC)
                 
                <span
                  class="visuallyHidden"
                >
                  British National Formulary for Children
                </span>
              </abbr>
              <span
                aria-hidden="true"
                class="tooltip"
              >
                British National Formulary for Children
              </span>
            </a>
          </li>
          <li>
            <a
              class="link"
              href="https://www.nice.org.uk/about/what-we-do/life-sciences"
            >
              Life sciences
            </a>
          </li>
          <li>
            <a
              class="link"
              href="https://www.nice.org.uk/library-and-knowledge-services"
            >
              Library and knowledge services
            </a>
          </li>
        </ul>
      </nav>
      <nav
        class="wrapper"
      >
        <ul
          class="list"
        >
          <li>
            <a
              href="https://www.nice.org.uk/what-nice-does"
            >
              What NICE does
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/implementing-nice-guidance"
            >
              Implementing NICE guidance
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/get-involved"
            >
              Get involved
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/about-us"
            >
              About us
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/careers"
            >
              Careers
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/news"
            >
              News, blogs and podcasts
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/nice-newsletters-and-alerts"
            >
              Newsletters and alerts
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/contact-us"
            >
              Contact us
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/forms/leave-feedback"
            >
              Leave feedback
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/reusing-our-content"
            >
              Reusing our content
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/forms/use-of-nice-content-in-the-uk"
            >
              NICE UK Open Content Licence
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/about/what-we-do/nice-syndication-api"
            >
              Syndicate our content
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/events"
            >
              Events
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="wrapper"
      >
        <ul
          aria-label="NICE on social media"
          class="list"
        >
          <li>
            <a
              href="https://en-gb.facebook.com/NationalInstituteforHealthandCareExcellence/"
            >
              <svg
                aria-hidden="true"
                class="icon"
                fill="none"
                focusable="false"
                height="1em"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M291.069 178.811H361l-8.168 77.197h-61.778V480h-92.789V256.008H152v-77.197h46.265V132.32c0-32.983 7.802-57.95 23.407-74.897C237.276 40.474 262.949 32 298.689 32h61.778v77.196h-38.646c-7.071 0-12.74.589-17.006 1.766-4.267 1.177-7.351 3.309-9.25 6.394-1.9 3.085-3.124 6.211-3.673 9.377-.549 3.167-.823 7.652-.823 13.457v38.621Z"
                  fill="currentColor"
                />
              </svg>
              Facebook
            </a>
          </li>
          <li>
            <a
              href="https://x.com/NICEcomms"
            >
              <svg
                aria-hidden="true"
                class="icon"
                fill="none"
                focusable="false"
                height="1em"
                viewBox="0 0 1200 1227"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M714.163 519.284 1160.89 0h-105.86L667.137 450.887 357.328 0H0l468.492 681.821L0 1226.37h105.866l409.625-476.152 327.181 476.152H1200L714.137 519.284h.026ZM569.165 687.828l-47.468-67.894-377.686-540.24h162.604l304.797 435.991 47.468 67.894 396.2 566.721H892.476L569.165 687.854v-.026Z"
                  fill="currentColor"
                />
              </svg>
              X
            </a>
          </li>
          <li>
            <a
              href="https://www.youtube.com/user/NICEmedia"
            >
              <svg
                aria-hidden="true"
                class="icon"
                fill="none"
                focusable="false"
                height="1em"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M209.753 314.5 330.746 252l-120.993-63.25V314.5ZM256 99c27.998 0 55.039.375 81.12 1.125 26.082.75 45.206 1.542 57.372 2.375l18.249 1c.167 0 1.584.125 4.25.375 2.667.25 4.583.5 5.75.75 1.166.25 3.125.625 5.874 1.125 2.75.5 5.125 1.167 7.125 2s4.333 1.917 7 3.25a48.044 48.044 0 0 1 7.749 4.875c2.5 1.917 4.917 4.125 7.25 6.625 1 1 2.291 2.542 3.875 4.625 1.583 2.083 3.999 6.958 7.249 14.625 3.25 7.667 5.458 16.083 6.625 25.25 1.333 10.667 2.375 22.042 3.125 34.125.75 12.083 1.208 21.542 1.375 28.375v44a529.498 529.498 0 0 1-4.5 72.5c-1.167 9.167-3.25 17.458-6.25 24.875-3 7.417-5.666 12.542-7.999 15.375l-3.5 4.25c-2.333 2.5-4.75 4.708-7.25 6.625-2.5 1.917-5.083 3.5-7.749 4.75-2.667 1.25-5 2.292-7 3.125-2 .833-4.375 1.5-7.125 2-2.749.5-4.749.875-5.999 1.125s-3.167.5-5.75.75-3.958.375-4.125.375C370.91 412.417 318.663 414 256 414c-34.498-.333-64.455-.875-89.87-1.625-25.415-.75-42.123-1.375-50.122-1.875l-12.25-1-8.999-1c-6-.833-10.541-1.667-13.624-2.5s-7.333-2.583-12.75-5.25c-5.416-2.667-10.124-6.083-14.124-10.25-1-1-2.291-2.542-3.875-4.625-1.583-2.083-4-6.958-7.25-14.625-3.249-7.667-5.457-16.083-6.624-25.25-1.333-10.667-2.375-22.042-3.125-34.125-.75-12.083-1.208-21.542-1.374-28.375v-44a529.323 529.323 0 0 1 4.5-72.5c1.166-9.167 3.25-17.458 6.249-24.875 3-7.417 5.666-12.542 8-15.375l3.5-4.25c2.332-2.5 4.749-4.708 7.249-6.625A48.018 48.018 0 0 1 69.26 111c2.666-1.333 5-2.417 6.999-3.25 2-.833 4.375-1.5 7.125-2s4.708-.875 5.874-1.125c1.167-.25 3.084-.5 5.75-.75s4.083-.375 4.25-.375C141.09 100.5 193.337 99 256 99Z"
                  fill="currentColor"
                />
              </svg>
              YouTube
            </a>
          </li>
          <li>
            <a
              href="https://www.linkedin.com/company/national-institute-for-health-and-clinical-excellence"
            >
              <svg
                aria-hidden="true"
                class="icon"
                fill="none"
                focusable="false"
                height="1em"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M134.024 181.394v289.59H37.565v-289.59h96.459Zm6.138-89.416c.196 14.226-4.725 26.112-14.764 35.658-10.039 9.546-23.243 14.319-39.612 14.319h-.589c-15.975 0-28.835-4.773-38.58-14.319C36.872 118.09 32 106.204 32 91.978c0-14.412 5.02-26.347 15.059-35.806C57.099 46.713 70.204 41.989 86.376 42c16.172.01 29.13 4.735 38.874 14.172 9.745 9.437 14.716 21.372 14.912 35.806Zm340.822 213.035V471H384.82V316.125c0-20.456-3.945-36.483-11.835-48.08-7.889-11.597-20.215-17.395-36.976-17.395-12.276 0-22.555 3.36-30.838 10.08-8.282 6.721-14.469 15.05-18.561 24.989-2.139 5.848-3.209 13.736-3.209 23.663v161.602h-96.164c.393-77.733.59-140.758.59-189.078 0-48.319-.099-77.154-.295-86.503l-.295-14.025h96.164v42.074h-.589c3.896-6.23 7.89-11.685 11.982-16.365 4.092-4.68 9.597-9.748 16.515-15.203 6.919-5.455 15.398-9.693 25.437-12.715 10.039-3.022 21.197-4.533 33.473-4.533 33.326 0 60.121 11.057 80.385 33.171C470.868 229.921 481 262.312 481 304.98l-.016.033Z"
                  fill="currentColor"
                />
              </svg>
              LinkedIn
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="container"
      >
        <nav
          aria-label="Legal menu"
          class="menu"
        >
          <ul>
            <li>
              <a
                href="https://www.nice.org.uk/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li>
              <a
                href="https://www.nice.org.uk/freedom-of-information"
              >
                Freedom of information
              </a>
            </li>
            <li>
              <a
                href="https://www.nice.org.uk/glossary"
              >
                Glossary
              </a>
            </li>
            <li>
              <a
                href="https://www.nice.org.uk/terms-and-conditions"
              >
                Terms and conditions
              </a>
            </li>
            <li>
              <a
                href="https://www.nice.org.uk/privacy-notice"
              >
                Privacy notice
              </a>
            </li>
            <li>
              <a
                href="https://www.nice.org.uk/cookies"
              >
                Cookies
              </a>
            </li>
          </ul>
        </nav>
        <p
          class="copyright"
        >
          ©
           
          <abbr
            title="National Institute for Health and Care Excellence"
          >
            NICE
          </abbr>
           
          2025
          . All rights reserved. Subject to
           
          <a
            href="https://www.nice.org.uk/terms-and-conditions#notice-of-rights"
          >
            Notice of rights
          </a>
          .
        </p>
      </div>
    </div>
  </footer>
</div>
`;
